storage/csi-snapshot.md

```diff
@@ -0,0 +1,115 @@
+* Container development benefits from filesystems that provide snapshot functionality.
+
+### Goals
+
+* Rebase the downstream csi-external-snapshotter and csi-external-provisioner images off of the upstream images based on Kubernetes 1.16.
+
+* Package and ship a downstream image of the this sidecar.
+
+### Non-Goals
+
+## Proposal
+
+This feature has already been implemented upstream, and is described in https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/20190709-csi-snapshot.md . It is currently documented upstream at https://kubernetes.io/docs/concepts/storage/volume-snapshots/ .
+
+We will be releasing the `external-snapshotter` sidecar image to enable CSI Snapshot support in OCP as Developer Preview only. This sidecar can be used by internal teams, such as CNV or OCS, for their drivers.
+
+### User Stories [optional]
+
+#### Story 1
+
+As an OCS developer, I want to release the `csi-external-snapshotter` sidecar image so that users can utilize volume snapshots.
+
+#### Story 2
+
+### Implementation Details/Notes/Constraints [optional]
+
+This feature has already been implemented upstream. Therefore, this feature requires a Kubernetes rebase of 1.15 or later.
+
+### Risks and Mitigations
+
+* Kubernetes 1.15+ rebase
+
+## Design Details
+
+No design is needed, as this feature has already been implemented upstream. The upstream design was discussed under https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/20190709-csi-snapshot.md.
+
+### Test Plan
+
+We’ll want the e2e tests to run using CSI sidecars shipped as part of OCP. There is ongoing progress in the following links regarding testing OCP CSI sidecars:
+
+* https://github.com/openshift/origin/pull/23560
+* https://jira.coreos.com/browse/STOR-223
+Tests for this sidecar are currently defined upstream at https://github.com/kubernetes-csi/external-snapshotter/tree/master/pkg/controller .
+
+### Graduation Criteria
+
+##### Dev Preview -> Tech Preview
+
+* Feature complete, including:
+  * Create/delete volume snapshots
+  * Create new volumes from a snapshot
+  * SnapshotContent Deletion/Retain Policy
+  * Snapshot Object in Use Protection
+  * Separate the common controller from the sidecar controller
+  * Add secrets field to list-snapshots RPC in the CSI spec. Add “data-source-secret” in create-volume intended for accessing the data source. Implement them in external-snapshotter and external-provisioner.
+  * Add metrics support
+* Unit and e2e tests implemented
+* Update snapshot CRDs to v1beta1 and enable VolumeSnapshotDataSource feature gate by default. The feature must also be at least v1beta1 upstream, and have a strong indication that it will be made GA in a reasonable time frame with a compatible API.
+
+##### Tech Preview -> GA
+
+* Snapshot feature is used as a basic building block in other advanced applications.
+* Feature deployed in production and have gone through at least one K8s upgrade.
+* Feature must be GA in Kubernetes.
```

Comment:
Do we want to drop this as well (like the other OEPs)?

Reply:
I've removed the sections that I don't believe are pertinent to OCP. I've left a couple of these, as I believe they're still relevant here, specifically regarding the promotion in upstream.

If I'm wrong, that's not a problem, and I can remove the remainder of these.